src/hashtable.py

```diff
@@ -0,0 +1,171 @@
+"""Hash table module...."""
+
+
+class HashTable(object):
+    """Hash table with multiple hashing functions."""
+
+    def __init__(self, size=1024, hash_func='additive'):
+        """Set up hash with specified hash function and size."""
+        self.nums = [10, 6, 3, 11, 15]
+        self.size = size
+        self.count = 0
+        self.buckets = []
+        self.hash_funcs = {
+            'additive': self._additive_hash,
+            'oat': self._oat_hash,
+            'oneatatime': self._oat_hash,
+            'fnv': self._fnv_hash,
+            'custom': self._custom_oat_hash
+        }
+
+        inpt = hash_func.lower().replace('-', '')
+        try:
+            self.hash_func = self.hash_funcs[inpt]
+        except KeyError:
+            raise NameError("Hash function, " + hash_func + ", is not an option.")
+
+        for i in range(size):
+            self.buckets.append([])
+
+    def get(self, key):
+        """Retrieve the given val using the given key."""
+        bucket = self._get_bucket(key)
+        idx = self._get_sub_bucket_idx(key, bucket)
+        if idx is not None:
+            return bucket[idx][1]
+        else:
+            raise KeyError('Key not found in hash table.')
+
+    def set(self, key, val):
+        """Store the given val using the given key."""
+        bucket = self._get_bucket(key)
+        idx = self._get_sub_bucket_idx(key, bucket)
+        if idx is not None:
+            bucket[idx] = (key, val)
+        else:
+            self.count += 1
+            bucket.append((key, val))
+
+    def _get_bucket(self, key):
+        idx = self._hash(key) % self.size
+        return self.buckets[idx]
+
+    def _get_sub_bucket_idx(self, key, bucket):
+        for idx, kv_pair in enumerate(bucket):
+            if kv_pair[0] == key:
+                return idx
+
+    def _hash(self, key):
+        """Hash the provided key."""
+        if type(key) is str:
+            return self.hash_func(key)
+        else:
```

Comment:
Nice, but a slightly more pythonic setup is

`` if something:
         return
    raise something```

The less indented the better :)